design/use-statement.md

```diff
@@ -0,0 +1,162 @@
+
+## Problems
+
+### Semantics of the task
+Some tools are hard to install on the fly.
+So, these are really "use a version from some limited subset of possible versions".
+Others can be used to fetch arbitrary versions from the internet.
+We introduced a split between "use version" and "install tool", but this has turned out to be more confusing than helpful in practice.
+
+### Naming
+The tasks are not named consistently, and they take inconsistent inputs.
+We have 2 `*Tool` tasks, 3 `*Installer` tasks, 2 `*ToolInstaller` tasks, and 2 `Use*Version` tasks.
+Some take a `version`, others a `versionSpec`, and one a `versionSelector`.
+See the appendix for a list of current tasks.
+
+### YAML syntax
+Some of the tasks take complex, interdependent inputs (Visual Studio Test Platform Installer).
+This is really hard to work with in YAML.
+Also, our `taskName@taskVersion` syntax adds a layer of confusion.
+```yaml
+- task: UsePythonVersion@0
+  inputs:
+    versionSpec: 3.x
+```
+The tasks are all at version 0 right now, so real-world confusion is *probably* limited.
+
+### Other things we want to include
+- Proxy setup.
+The agent knows about its proxy; the tasks don't always pass that along to client tooling.
+- Auth setup.
+If I'm going to publish to a private npm registry, I should be able to give the creds here.
+- Problem matchers.
+If the Python compiler emits errors in a known format, we should be able to read that via regex and surfaec it in our UI.
+VS Code has a defined format for problem matchers.
+
+## Solution
+
+We'll add new YAML syntax which puts tools on the `PATH`, adds proxy information, adds problem matchers, and optionally configures auth.
+This syntax will be powered by tasks, giving a familiar extensibility point for third parties to work with.
+
+### New syntax
+
+YAML will get a new syntax for `use`-ing a tool or ecosystem.
+```yaml
+- use: {toolName} {versionSpec}
+```
+
+This can be implemented as sugar over the existing syntax, much like the `powershell` and `bash` keywords today.
+
+`{versionSpec}` is optional, and not needed in some cases (see below).
+It can also be passed as an explicit parameter, like this:
+```yaml
+- use: {toolName}
+  version: {versionSpec}
+```
+
+By default, `use` will set up the target ecosystem to use the agent's proxy.
+If this behavior isn't desired, it can be overridden with:
+```yaml
+- use: someTool someVersion
+  proxy: false
+```
+
+If a tool needs credentials, those can be added like this:
+```yaml
+resources:
+  feeds:
+  - feed: myFeed
+    auth: {auth_information}
+
+steps:
+- use: someTool someVersion
+  auth: myFeed
+```
+
+`feed` is not yet a resource type, but it's coming.
+
+No additional keywords are reserved.
+Any additional keywords are passed through as `inputs` to the underlying task.
+
+## Schema
+
+```yaml
+- use: string       # required tool name and optional version
+  version: string   # optional version (specifying version twice is an error)
```

Comment:
Which input does this map to? Does this mean the task must name the input `version`? Or does this mean they need to go add an alias to the input (we support that today).